patch-package
--- a/node_modules/web3-core-requestmanager/src/batch.js
+++ b/node_modules/web3-core-requestmanager/src/batch.js
@@ -53,16 +53,22 @@ Batch.prototype.execute = function () {
             return results[index] || {};
         }).forEach(function (result, index) {
             if (requests[index].callback) {
+                var cb = requests[index].callback;
 
                 if (result && result.error) {
-                    return requests[index].callback(errors.ErrorResponse(result));
+                    return cb(errors.ErrorResponse(result));
                 }
 
                 if (!Jsonrpc.isValidResponse(result)) {
-                    return requests[index].callback(errors.InvalidResponse(result));
+                    return cb(errors.InvalidResponse(result));
+                }
+
+                try {
+                    cb(null, requests[index].format ? requests[index].format(result.result) : result.result);
+                } catch (err) {
+                    cb(err);
                 }
 
-                requests[index].callback(null, (requests[index].format ? requests[index].format(result.result) : result.result));
             }
         });
     });
